controllers/awscluster_controller.go

```diff
@@ -252,6 +263,44 @@ func (r *AWSClusterReconciler) reconcileDelete(ctx context.Context, clusterScope
 	return nil
 }
 
+func (r *AWSClusterReconciler) reconcileLoadBalancer(clusterScope *scope.ClusterScope, awsCluster *infrav1.AWSCluster) (*time.Duration, error) {
+	if annotations.Has(awsCluster, DisableLoadBalancerReconciliationAnnotation) {
+		clusterScope.Debug("load balancer reconciliation shifted to external provider, checking external endpoint")
+
+		return r.checkForExternalControlPlaneLoadBalancer(clusterScope, awsCluster), nil
+	}
+
```

Comment:
We don't know if the externally managed control plane will create its own IP address, as e.g. Kamaji is going, or rather because the user is hard-coding a pre-defined IP address in the `/AWSCluster/spec/controlPlaneEndpoint` field.

In such edge-case scenarios, the knob through the annotation will skip the creation of the infrastructure components that would be unnecessary.

If you prefer, we could rework the PR and embed this logic in the retrieval of the Control Plane object: if it's marked as externally managed, we can skip the CP infrastructure components.

Reply:
Fixed by adding a field to the `AWSCluster` type, which made this outdated.